distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -193,8 +194,55 @@ type Date
          other hand, the first day of the week is Monday, and week 1 is the week
          containing the first Thursday of the year. Therefore it is important to
          properly specify the `locale` argument.
-    week_of_year : Locale.Locale -> Integer
-    week_of_year self locale=Locale.default = Time_Utils.week_of_year_localdate self locale.java_locale
+    week_of_year : (Locale.Locale | Nothing) -> Integer
+    week_of_year self locale=Nothing =
+        if locale.is_nothing then
+            Time_Utils.week_of_year_localdate self
+        else
+            Time_Utils.week_of_year_localdate self locale.java_locale
+
+
+    ## Returns if the date is in a leap year.
+    is_leap_year : Boolean
+    is_leap_year self =
+        year = self.year
+        (year%4 == 0) && ((year%100 != 0) || (year%400 == 0))
+
+    ## Returns the number of days in the year represented by this date.
+    length_of_year : Integer
+    length_of_year self =
+        if self.is_leap_year then 366 else 365
+
+    ## Returns the century of the date.
+    century : Integer
+    century self = (self.year - 1).div 100 + 1
+
+    ## Returns the quarter of the year the date falls into.
+    quarter : Integer
+    quarter self = (self.month - 1).div 3 + 1
+
+    ## Returns the number of days in the month represented by this date.
+    length_of_month : Integer
+    length_of_month self =
+        case self.month of
+            2 -> if self.is_leap_year then 29 else 28
+            4 -> 30
+            6 -> 30
+            9 -> 30
+            11 -> 30
+            _ -> 31
```

Comment:
Same as above, can't we use https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/time/LocalDate.html#lengthOfMonth() ?